File: NQT/NumberSystem/DecimalToOctal.java
package NumberSystem;

public class DecimalToOctal {
    public static void main(String[] args) {
        int decimalNumber = 45; // You can change this value to test with different decimal numbers
        String octalValue = decimalToOctal(decimalNumber);
        System.out.println("The octal value of decimal " + decimalNumber + " is: " + octalValue);
    }

    private static String decimalToOctal(int decimal) {
        if (decimal == 0) {
            return "0";
        }
        StringBuilder octalValue = new StringBuilder();
        while (decimal > 0) {
            int remainder = decimal % 8;
            octalValue.insert(0, remainder); // Prepend the remainder to the octal string
            decimal /= 8; // Divide the decimal number by 8
        }
        return octalValue.toString();
    }
}

File: NQT/NumberSystem/decimal_to_octal.py
number1 = 45
octal = ""
# convert a decimal number to octal without using built-in functions
number1 = int(number1)
while number1 > 0:
    remainder = number1 % 8
    octal = str(remainder) + octal
    number1 //= 8
print(octal)  # Output: 55

# using built-in function
number1 = 45
octal = oct(number1)[2:]  # Convert decimal to octal and remove '0o'
print(octal)  # Output: 55